packages/block-editor/src/components/inserter/search-form.native.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import { TextInput, View, TouchableHighlight } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useRef } from '@wordpress/element';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+import { ToolbarButton } from '@wordpress/components';
+import {
+	Icon,
+	cancelCircleFilled,
+	arrowLeft,
+	search as searchIcon,
+} from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+function InserterSearchForm( { value, onChange, onLayout = () => {} } ) {
+	const [ isActive, setIsActive ] = useState( false );
+
+	const inputRef = useRef();
+
+	const searchFormStyle = usePreferredColorSchemeStyle(
+		styles.searchForm,
+		styles.searchFormDark
+	);
+
+	const searchFormInputStyle = usePreferredColorSchemeStyle(
+		styles.searchFormInput,
+		styles.searchFormInputDark
+	);
+
+	const placeholderStyle = usePreferredColorSchemeStyle(
+		styles.searchFormPlaceholder,
+		styles.searchFormPlaceholderDark
+	);
+
+	return (
+		<TouchableHighlight accessible={ false } onLayout={ onLayout }>
+			<View style={ searchFormStyle }>
+				{ isActive ? (
+					<ToolbarButton
+						title={ __( 'Cancel search' ) }
+						icon={ arrowLeft }
+						onClick={ () => {
+							inputRef.current.blur();
+							onChange( '' );
```

Comment:
Out of curiousity, why default to empty string instead of undefined?

Reply:
Good question, I was thinking the same when I looked back at this PR 😆 . After reviewing the changes as they are now I can't find an explicit reason. I believe as some point while I was drafting the changes I was treating `undefined` differently. That's not the case now.

I might update the `onChange`  function definition to use `''` as the default value in a later change. That way the value remains a  `String` type but can be called with `undefined`